### DISCUSSION TITLE
Implicit Type conversion for generics in instance declaration

### DISCUSSION THREAD

**Discussion by 80O** [dotnet/csharplang#2689] [Language Ideas]
I'm proposing to have implicit type conversion in instance declaration for generics like Java.

**For example**:

`LinkedList<string> names = new LinkedList<string>();`

**Becomes**:

`LinkedList<string> names = new LinkedList<>();`

I often find myself wondering why C# does not allow this and needs explicitly the type declared in the new instance while it can be inferred from the type declaration already. 

The <> should be required as otherwise the type is different and no longer generic.

**Comment by DaZombieKiller** [comment]
In the case of a variable, I would just use
`var names = new LinkedList<string>();`

For a field, you can use target-typed `new` from C# 8
`LinkedList<string> names = new();`

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/issues/1349
